Styra/Opa/OpaBatchTypes.cs

```diff
@@ -0,0 +1,74 @@
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Styra.Opa.OpenApi.Models.Components;
+
+namespace Styra.Opa;
+
+public class OpaBatchInputs : Dictionary<string, Dictionary<string, object>>
+{
+    public override string ToString()
+    {
+        return JsonConvert.SerializeObject(this);
+    }
+}
+
+public class OpaBatchResults : Dictionary<string, OpaResult>
+{
+    public override string ToString()
+    {
+        return JsonConvert.SerializeObject(this);
+    }
+}
+
+public class OpaBatchErrors : Dictionary<string, OpaError>
+{
+    public override string ToString()
+    {
+        return JsonConvert.SerializeObject(this);
+    }
+}
+
+// Used for converting inputs for the Batch Query API, and converting result types
+// into useful higher-level types.
+public static class DictionaryExtensions
+{
+    public static Dictionary<string, Input> ToOpaBatchInputRaw(this Dictionary<string, Dictionary<string, object>> inputs)
+    {
+        var opaBatchInputs = new Dictionary<string, Input>();
+        foreach (var kvp in inputs)
+        {
+            opaBatchInputs[kvp.Key] = Input.CreateMapOfAny(kvp.Value);
+        }
+        return opaBatchInputs;
+    }
+
+    public static OpaBatchErrors ToOpaBatchErrors(this Dictionary<string, Styra.Opa.OpenApi.Models.Errors.ServerError> errors)
+    {
+        var opaBatchErrors = new OpaBatchErrors();
+        foreach (var kvp in errors)
+        {
+            opaBatchErrors[kvp.Key] = new OpaError(kvp.Value);
+        }
+        return opaBatchErrors;
+    }
+
+    public static OpaBatchErrors ToOpaBatchErrors(this Dictionary<string, Styra.Opa.OpenApi.Models.Components.ServerError> errors)
+    {
+        var opaBatchErrors = new OpaBatchErrors();
+        foreach (var kvp in errors)
+        {
+            opaBatchErrors[kvp.Key] = new OpaError(kvp.Value);
+        }
+        return opaBatchErrors;
+    }
+
+    public static OpaBatchResults ToOpaBatchResults(this Dictionary<string, SuccessfulPolicyResponse> responses)
+    {
+        var opaBatchResults = new OpaBatchResults();
+        foreach (var kvp in responses)
+        {
+            opaBatchResults[kvp.Key] = (OpaResult)kvp.Value;
+        }
+        return opaBatchResults;
+    }
+}
```

Comment:
Missing newline at EOF (same for most files).

Reply:
Rolled into latest changes. Thanks for noticing that inconsistency!